tests/components/aurora_abb_powerone/test_config_flow.py

```diff
@@ -0,0 +1,155 @@
+        return_value=fakecomports,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch("aurorapy.client.AuroraSerialClient.connect", return_value=None,), patch(
+        "aurorapy.client.AuroraSerialClient.serial_number",
+        return_value="9876543",
+    ), patch(
+        "aurorapy.client.AuroraSerialClient.version",
+        return_value="9.8.7.6",
+    ), patch(
+        "aurorapy.client.AuroraSerialClient.pn",
+        return_value="A.B.C",
+    ), patch(
+        "aurorapy.client.AuroraSerialClient.firmware",
+        return_value="1.234",
+    ), patch(
+        "homeassistant.components.aurora_abb_powerone.config_flow._LOGGER.getEffectiveLevel",
+        return_value=INFO,
+    ), patch(
+        "homeassistant.components.aurora_abb_powerone.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.aurora_abb_powerone.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_PORT: "/dev/ttyUSB7", CONF_ADDRESS: 7},
+        )
+
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+
+    assert result2["data"] == {
+        CONF_PORT: "/dev/ttyUSB7",
+        CONF_ADDRESS: 7,
+        ATTR_FIRMWARE: "1.234",
+        ATTR_MODEL: "9.8.7.6 (A.B.C)",
+        ATTR_SERIAL_NUMBER: "9876543",
+        "title": "PhotoVoltaic Inverters",
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_no_comports(hass):
+    """Test we display correct info when there are no com ports.."""
+
+    fakecomports = []
+    with patch(
+        "serial.tools.list_ports.comports",
+        return_value=fakecomports,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+    print("result=%s" % result)
+    assert result["type"] == "abort"
+    assert result["reason"] == "no_serial_ports"
+
+
+async def test_form_invalid_com_ports(hass):
+    """Test we display correct info when the comport is invalid.."""
+
+    fakecomports = []
+    fakecomports.append(list_ports_common.ListPortInfo("/dev/ttyUSB7"))
+    with patch(
+        "serial.tools.list_ports.comports",
+        return_value=fakecomports,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+    print("result=%s" % result)
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "aurorapy.client.AuroraSerialClient.connect",
+        side_effect=OSError("...no such device..."),
```

Comment:
Mock the OSError code instead of the message

Reply:
Agreed. Done.